src/assets/scripts/client/airline/AirlineModel.js

```diff
@@ -218,29 +206,23 @@ export default class AirlineModel extends BaseModel {
      */
     generateFlightNumber() {
         let flightNumber = '';
-        let list = '0123456789';
-
-        // Start with a number other than zero
-        flightNumber += choose(list.substr(1));
-
-        if (!this.flightNumberGeneration.alphaNumeric) {
-            for (let i = 1; i < this.flightNumberGeneration.length; i++) {
-                flightNumber += choose(list);
+        const numbersList = '0123456789';
+        const alphaList = 'abcdefghijklmnopqrstuvwxyz';
+        const formats = this.flightNumberGeneration.callsignFormats;
+
+        const chosenFormat = choose(formats);
+        for (let i = 0; i < chosenFormat.length; i++) {
+            if (chosenFormat[i] === '#') {
```

Comment:
these nested `if`s and `else`s make me cringe.  there is a better, cleaner way to do this that I'll post here later.